File: Python_Programming/Arithmetic_Pattern.py


def Arithmetic_Pattern(Rows,Cols):

    if(Rows!=Cols):
        print("Invalid number of rows or columns")
        return

    for i in range(1,Rows+1):
        for j in range(1,Cols+1):

            if((i==(Cols//2)+1) or ((j==(Rows//2)+1))):
                print("*",end="")
            else:
                print(" ",end="")

        print("")

    print("")
    
    for i in range(1,Rows+1):
        for j in range(1,Cols+1):
            if(i==(Rows//2)):
                print("*",end="")
            else:
                print(" ",end="")
        print("")

    
            
    for i in range(1,Rows+1):
        for j in range(1,Cols+1):
            if(i==j or j==Rows-i+1):
                print("*",end="")
            else:
                print(" ",end="")
        print("")

    print("")
    print("")

    for i in range(1,Rows+1):
        for j in range(1,Cols+1):
            if(i+j==Rows+1):
                print("*",end="")
            else:
                print(" ",end="")

        print("")


def main():
    Rows=0
    Cols=0

    Rows=int(input("Enter number of Rows : "))
    Cols=int(input("Enter number of Columns : "))

    Arithmetic_Pattern(Rows,Cols)

if __name__=="__main__":
    main()
    

File: C++_Programming/Arithmetic_Pattern.cpp
#include<iostream>
using namespace std;



void Arithmetic_Pattern(int Rows,int Cols)
{
    int i=0;
    int j=0;

    if(Rows!=Cols)
    {
        printf("Invalid number of rows or columns\n");
        return;
    }

    for(i=1;i<=Rows;i++)
    {
        for(j=1;j<=Cols;j++)
        {
            if((i==(Cols/2)+1) ||((j==(Rows/2)+1)))
            {
                cout<<"*";
            }
            else
            {
                cout<<" ";
            }

            
        }
        cout<<"\n";
    }

    printf("\n");
    
    for(i=1;i<=Rows;i++)
    {
        for(j=1;j<=Cols;j++)
        {
            if(i==(Rows/2))
            {
                cout<<"*";
            }
            else
            {
                cout<<" ";
            }

            
        }
        cout<<"\n";
    }

    
    for(i=1;i<=Rows;i++)
    {
        for(j=1;j<=Cols;j++)
        {
            if(i==j || j==Rows-i+1)
            {
                cout<<"*";
            }
            else
            {
                cout<<" ";
            }
        }
        cout<<"\n";
    }

    cout<<"\n";
    cout<<"\n";

    for(i=1;i<=Rows;i++)
    {
        for(j=1;j<=Cols;j++)
        {
            if(i+j==Rows+1)
            {
                cout<<"*";
            }
            else
            {
                cout<<" ";
            }

            
        }
        cout<<"\n";
    }
}

int main()
{
    int Rows=0;
    int Cols=0;

    cout<<"Enter number of rows : ";
    cin>>Rows;
    cout<<"Enter number of columns : ";
    cin>>Cols;

    Arithmetic_Pattern(Rows,Cols);

    return 0;
}